### ISSUE TITLE
Unable to use keybase recently, certificate errors

### ISSUE THREAD

**Issue by cvervais-splunk** [keybase/keybase-issues#3828]
Hey folks,

   Recently, in the past couple of weeks, I can no longer load new chat messages on either my iOS client or MacOS client.

   I was attempting to do a re-install of the client on my MacOS system and also to reset the passphrase since I forgot it but, do have my paper keys but, not even that is working. I get errors about a certificate mismatch.....

`Chris-iMac:~ cvervais$ keybase passphrase recover
Your keybase username: cvervais
▶ INFO Failed to check server for revoked in GAPU: API network error: doRetry failed, attempts: 3, timeout 5s, last err: Post https://api-0.core.keybaseapi.com/_/api/1.0/device/for_users.json: x509: certificate is valid for *.testing.sp2.ring.net, not api-0.core.keybaseapi.com [tags:GAPU=Zs3AU9kJmu-f]

▶ ERROR API network error: doRetry failed, attempts: 3, timeout 5.324s, last err: Get https://api-0.core.keybaseapi.com/_/api/1.0/merkle/path.json?c=1&last=14600155&load_deleted=1&load_reset_chain=1&poll=10&sig_hints_low=4&uid=0015bee3fed4b5d9f5b8234dee5d8519: x509: certificate is valid for *.testing.sp2.ring.net, not api-0.core.keybaseapi.com (code 1601)`

I also get the same error when trying to reset the passphrase from the client.

I'm wondering if the errors I'm getting above could be related to my inability to load recent chat messages.

**Comment by cvervais-splunk** [comment]
Though I'm also concerned your endpoints are showing a ring certificate

**Comment by cvervais-splunk** [comment]
Alright, I figured this out. And, it was bizarre. 

tl;dr looks like the DNS cache on my router was holding on to a stale entry, a reboot of the router fixed it.

The two major clues were, the certificate being presented when connecting to the API being a cert belonging to RING and the IP addresses being retuned for api-0.core.keybaseapi.com being AWS IP addresses.

As I'm sure many know, classic load balancers in AWS don't have static IPs on them and change regularly. Meaning that a IP fronting the keybase API one day might be fronting someone elses API the next day. 

In this case, when the IPs fronting the keybase API changed, the DNS service on my router for whatever reason ignored the DNS TTL and kept the IP addresses cached. The IPs were now fronting a ring service hence why the ring cert was being presented. 

A clue on this was that when my phone was off my network, keybase was working. Doing a straight A/B test made it clear the problem was local to my network. 

From my network, querying googles DNS servers and OpenDNS server got a different response than querying my standard resolver. Even clearing the DNS cache on the MacOS system didn't have an effect therefore the problem was upstream. Hence either my router or, my ISPs resolvers. 

Since I have more control over my router than the ISPs resolvers, I decided to focus on the router first and reboot it to see if problem went away. And, it did, so, happy times.

Posting this lengthy explanation so other folks that land here don't see a dead thread with the last comment showing concern why a ring cert was showing up on a keybase endpoint.
